### ISSUE TITLE
"list" syntax does not support "." as argument.

### ISSUE THREAD

**Issue by yum-feng** [gotcha/ipdb#203]
The command `l` will print the source code based on where it last left off, which is as expected. In the normal `pdb` debugger, I would specify `.` as argument, `l .` to get back to the current line of execution. 

This does not seem to work in IPDB, or am I doing something wrong ?

I'm running `Python 3.6.8` with the following versions.

`ipdb==0.13.3`
`ipython==7.14.0`
`ipython-genutils==0.2.0`

**Comment by willhansen** [comment]
I've also run into this.  It feels like a bug, because the only other way to reset `list` I've found is to use `w` to see the current line number, then use that as an argument for `list`

**Comment by gotcha** [comment]
I guess this is an issue with IPython. Can you check that you have the issue without ``ipdb`` ?

**Comment by wztdream** [comment]
same issue here, and confirm **pdb** support `l .`

**Comment by mzr1996** [comment]
Same issue.

**Comment by isidroas** [comment]
Same issue, with ipdb it raises an error,
![2022-11-27-105741_](https://user-images.githubusercontent.com/10489506/204129317-40b41a3f-e72d-4f1e-b3f6-c2b09094bc94.png)

but with pdb, it isn't,
![2022-11-27-105040_](https://user-images.githubusercontent.com/10489506/204129107-658232df-1b4c-4f45-9d86-6b5d44b46c62.png)

IPython version: 8.4.0
Python version: 3.10.6
ipdb version: 0.13.9

**Comment by jessie-hu-95** [comment]
I found a workaround to support this. This definitely is an issue of IPython, and we can figure this out in their source code.
By comparing `do_list()` in [IPython](https://github.com/ipython/ipython/blob/f112764278c3c4401b45a4b71ac1461801cab93d/IPython/core/debugger.py#L638) and the function with same name in [PDB](https://github.com/python/cpython/blob/d2340ef25721b6a72d45d4508c672c4be38c67d3/Lib/pdb.py#L1287), we can the difference (only for part relative to `'.'`), in IPython code they did not treat `'.'` as a valid argument.
To enable the `'.'` argument, one can apply the following patch:
```
diff --git a/IPython/core/debugger.py b/IPython/core/debugger.py
index c8082e34e..5b793d783 100644
--- a/IPython/core/debugger.py
+++ b/IPython/core/debugger.py
@@ -640,7 +640,7 @@ def do_list(self, arg):
         """
         self.lastcmd = 'list'
         last = None
-        if arg:
+        if arg and arg != '.':
             try:
                 x = eval(arg, {}, {})
                 if type(x) == type(()):
@@ -655,7 +655,7 @@ def do_list(self, arg):
             except:
                 print('*** Error in argument:', repr(arg), file=self.stdout)
                 return
-        elif self.lineno is None:
+        elif self.lineno is None or arg == '.':
             first = max(1, self.curframe.f_lineno - 5)
         else:
             first = self.lineno + 1
```

**Comment by gotcha** [comment]
@jessie-hu-95 Did you submit this as a pull request in the IPython repository ?

**Comment by jessie-hu-95** [comment]
I just created one.  But I do not know if my PR is created correctly, since this is my first PR.  🤔

**Comment by gotcha** [comment]
Should be released in IPython 8.15
